org.opent2t.sample.hub.superpopular/com.wink.hub/js/thingTranslator.js

```diff
@@ -89,6 +90,88 @@ class Translator {
+        // subscribing to the channels.  The subscribe_key will be the same for all
+        // devices on an account (even non Wink)
+        return this._makeRequest(this._userPath, 'GET').then((response) => {
+            this._pubnub = new PubNub({
+                subscribeKey: response.data.subscription.pubnub.subscribe_key
+            });
+
+            this._pubnub.addListener({
+                message: this._translateEvent.bind(this)
+            });
+
+            // Subscribe to the user channel for device discovery
+            pubnubChannels.push(response.data.subscription.pubnub.channel)
+
+            // Get all of the devices attached to this provider
+            // This can include non-Wink devices that have been connected to a Wink account
+            // such as a Nest Thermostat, though they will still be translated into a Wink representation
+            // by the Wink API.
+            return this._makeRequest(this._devicesPath, 'GET').then((response) => {
+                response.data.forEach((winkDevice) => {
+                    var opent2tInfo = this._getOpent2tInfo(winkDevice);
+
+                    // SKip any devices that do not have an opent2t definition
+                    if (opent2tInfo != undefined) {
+                        if (winkDevice.subscription != undefined) {
+                            pubnubChannels.push(winkDevice.subscription.pubnub.channel);
+                        }
+                    }
+                });
+
+                // Subscribe to all of the channels
+                this._pubnub.subscribe({ channels: pubnubChannels });
+            });
+        });
+    }
+
+    /**
+     * Translates realtime notifications from the appropriate Wink device schema to OCF and calls the
+     * registered callback function
+     */
+    _translateEvent(event) {
+        if (this._eventCallback !== undefined) {
+            var deviceData = JSON.parse(event.message);
+            var opent2tinfo = this._getOpent2tInfo(deviceData);
+
+            // Find the translator for this device
+            if (opent2tinfo !== undefined) {
+                var deviceInfo = {
+                    hub: this,
+                    deviceInfo: { id: 0 } // Device identifier isn't needed as the translator will not be used to
+                                          // communicate with an API endpoint.
+                };
+
+                // TODO: There must be a better way of doing this that escapes me at the moment
+                // We're not truly creating a translator, I just want access to the 'deviceSchemaToTranslatorSchema'
+                // function
+                var translatorName = opent2tinfo.translator.replace(/opent2t-translator-/gi, "").replace(/-/g, ".");
+                var translatorPath = require('path').join(__dirname, '../../..', opent2tinfo.schema, translatorName, "js", "thingTranslator.js");
```

Comment:
>var translatorPath = require('path').join(__dirname, '../../..', opent2tinfo.schema, translatorName, "js", "thingTranslator.js"); [](start = 16, length = 129)

relative paths may get you in to trouble, use OpenT2T.createTranslatorAsync to do this.